test/testdrive/kafka-avro-debezium-transaction.td

```diff
@@ -0,0 +1,345 @@
+        "default": null
+      },
+      {
+        "name": "data_collections",
+        "type": [
+          "null",
+          {
+            "type": "array",
+            "items": {
+              "type": "record",
+              "name": "ConnectDefault",
+              "namespace": "io.confluent.connect.Avro",
+              "fields": [
+                {"name": "data_collection", "type": "string"},
+                {"name": "event_count", "type": "long"}
+              ]
+            }
+          }
+        ],
+        "default": null
+      }
+    ],
+    "connect.name": "io.debezium.connector.common.TransactionMetadataValue"
+  }
+
+$ set txschema-bad-schema={
+    "type": "record",
+    "name": "TransactionMetadataValue",
+    "namespace": "io.debezium.connector.common",
+    "fields": [
+      {"name": "status", "type": "string"},
+      {
+        "name": "id",
+        "type": ["null", "string"]
+      },
+      {
+        "name": "event_count",
+        "type": ["null", "long"],
+        "default": null
+      },
+      {
+        "name": "data_collections",
+        "type": [
+          "null",
+          {
+            "type": "array",
+            "items": {
+              "type": "record",
+              "name": "ConnectDefault",
+              "namespace": "io.confluent.connect.Avro",
+              "fields": [
+                {"name": "data_collection", "type": "string"},
+                {"name": "event_count", "type": "long"}
+              ]
+            }
+          }
+        ],
+        "default": null
+      }
+    ],
+    "connect.name": "io.debezium.connector.common.TransactionMetadataValue"
+  }
+
+$ kafka-create-topic topic=data-txdata
+
+> CREATE MATERIALIZED SOURCE data_txdata
+  FROM KAFKA BROKER '${testdrive.kafka-addr}' TOPIC 'testdrive-data-txdata-${testdrive.seed}'
+  FORMAT AVRO USING SCHEMA '${txschema}'
+  ENVELOPE NONE
+
+
+$ kafka-ingest format=avro topic=data-txdata schema=${txschema} timestamp=4
+{"status": "BEGIN", "id": "1", "event_count": null, "data_collections": null}
+{"status": "END", "id": "1", "event_count": {"long": 3}, "data_collections": {"array": [{"event_count": 3, "data_collection": "testdrive-data-${testdrive.seed}"}]}}
+
+
+$ kafka-create-topic topic=data
+
+$ kafka-ingest format=avro topic=data schema=${schema} timestamp=1
+{"before": null, "after": {"row": {"a": 1, "b": 1}}, "source": {"file": "binlog", "pos": 0, "row": 0, "snapshot": {"boolean": false}}, "transaction": {"total_order": null, "id": "1"}}
+{"before": null, "after": {"row": {"a": 2, "b": 3}}, "source": {"file": "binlog", "pos": 1, "row": 0, "snapshot": {"boolean": false}}, "transaction": {"total_order": null, "id": "1"}}
+
+#
+# Create a source using an inline schema.
+#
+> CREATE MATERIALIZED SOURCE data_schema_inline
+  FROM KAFKA BROKER '${testdrive.kafka-addr}' TOPIC 'testdrive-data-${testdrive.seed}'
+  WITH(tx_metadata = data_txdata)
+  FORMAT AVRO USING SCHEMA '${schema}'
+  ENVELOPE DEBEZIUM
```

Comment:
Thoughts on where the new option lives syntactically?  Is it better suited being something like `ENVELOPE DEBEZIUM TRANSACTION_METADATA $name_of_source`?

Reply:
I think explicit keywords like this are better than `WITH`!